### PR TITLE
[Square] New Components

### LINKED ISSUE #3791: Square
### App
Square

### Triggers
* New Appointment
* New Catalog Item
* New Customer
* New Invoice
* New Order

### Actions
* Add Customer to Group
* Create Appointment
* Create Customer
* Create Invoice
* Create Order
* Remove Customer From Group
* Send Invoice
* Update Customer
* Find Customer
* Find Service
* Find Order

### PR DESCRIPTION
**Pull request by andrewjschuang**
To test this you need to create a Sandbox app and use the https://connect.squareupsandbox.com/v2 url.

**Actions:**

- Create Customer
- Create Invoice
- Create Order
- Send Invoice

**Triggers:**

- Catalog Item Updated
- New Booking Created
- New Customer Created
- New Invoice Created
- New Order Created
- Custom Webhook Events

[API Reference](https://developer.squareup.com/reference/square)
[API Explorer](https://developer.squareup.com/explorer/square)
[Docs / Guides](https://developer.squareup.com/docs)

Resolves #3791.

### CODE REVIEW

File: components/square/actions/create-invoice/create-invoice.mjs [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+            card: true,
+            bank_account: true,
+          },
+          primary_recipient: {
+            customer_id: this.customer,
+          },
+        },
+      },
+    });
+    $.export("$summary", "Successfully created invoice");
```
Comment (alysonturing) [inline components/square/actions/create-invoice/create-invoice.mjs:66]: You could export the Id of the generated resource.

File: components/square/actions/create-invoice/create-invoice.mjs [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+
+    const response = await this.square.createInvoice({
+      $,
+      generateIdempotencyKey: true,
+      data: {
+        invoice: {
+          location_id: this.location,
+          order_id: this.order,
+          payment_requests: paymentRequests,
+          delivery_method: "EMAIL",
```
Comment (alysonturing) [inline components/square/actions/create-invoice/create-invoice.mjs:55]: you could move it to the constants file

File: components/square/actions/create-invoice/create-invoice.mjs [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+    dueDate: {
+      type: "string",
+      label: "Due Date",
+      description: "The due date for the payment request, in `YYYY-MM-DD` format",
+    },
+  },
+  async run({ $ }) {
+    const paymentRequests = [
+      {
+        request_type: "BALANCE",
```
Comment (alysonturing) [inline components/square/actions/create-invoice/create-invoice.mjs:41]: you could move it to the constants file

File: components/square/actions/create-invoice/create-invoice.mjs [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+        automatic_payment_source: "NONE",
```
Comment (alysonturing) [inline components/square/actions/create-invoice/create-invoice.mjs:42]: you could move it to the constants file

File: components/square/sources/catalog-item-updated/catalog-item-updated.mjs [hunk 5]
```diff
@@ -0,0 +1,46 @@
+import base from "../common/base.mjs";
+import constants from "../common/constants.mjs";
+
+export default {
+  ...base,
+  key: "square-catalog-item-updated",
+  name: "Catalog Item Updated",
+  description: "Emit new event for every updated catalog item",
```
Comment (alysonturing) [inline components/square/sources/catalog-item-updated/catalog-item-updated.mjs:8]: is nice to have the link for the webhook docs

File: components/square/sources/new-booking-created/new-booking-created.mjs [hunk 6]
```diff
@@ -0,0 +1,45 @@
+import base from "../common/base.mjs";
+import constants from "../common/constants.mjs";
+
+export default {
+  ...base,
+  key: "square-new-booking-created",
+  name: "New Booking Created",
+  description: "Emit new event for every new booking created",
```
Comment (alysonturing) [inline components/square/sources/new-booking-created/new-booking-created.mjs:8]: is nice to have the link for the webhook docs

File: components/square/sources/new-order-created/new-order-created.mjs [hunk 7]
```diff
@@ -0,0 +1,67 @@
...
+              sort_field: "CREATED_AT",
+              sort_order: "DESC",
+            },
+          },
+          location_ids: [
+            this.location,
+          ],
+        },
+      });
+      response?.objects?.slice(-constants.MAX_HISTORICAL_EVENTS)
```
Comment (alysonturing) [inline components/square/sources/new-order-created/new-order-created.mjs:41]: this instruction is repeated over the sources and could be reused by a inherited method

**Review by alysonturing — DISMISSED** [review]
LGTM! 
But I added some opinions that you may want to check, and if it make sense to you, it may improve your contribution

**Comment by vunguyenhung** [comment]
Hello everyone, I have tested this PR and there're some test cases failed or needed improvement. 

Please check the test report below for more information
[Square_3791_2038.pdf](https://github.com/PipedreamHQ/pipedream/files/10314482/Square_3791_2038.pdf)

**Comment by andrewjschuang** [comment]
Hello @vunguyenhung, I couldn't confirm through the report, but did you test the components in a [sandbox environment](https://developer.squareup.com/docs/devtools/sandbox/overview) as mentioned above? The pending scopes were requested [here](https://github.com/PipedreamHQ/pipedream/issues/3791#issuecomment-1362848258) and will be needed for a production environment.

**Comment by vunguyenhung** [comment]
Hello everyone, I have tested this PR and there're some test cases failed or needed improvement. 

Please check the test report below for more information
[Square_3791_1440.pdf](https://github.com/PipedreamHQ/pipedream/files/10318175/Square_3791_1440.pdf)

**Comment by andrewjschuang** [comment]
Hello @vunguyenhung, thanks for testing! Issues should be fixed now.

**Comment by vunguyenhung** [comment]
Hi everyone, all test cases are passed! Ready for release! 

Test report
[Square_3791.pdf](https://github.com/PipedreamHQ/pipedream/files/10334593/Square_3791.pdf)

**Pull request merged**